src/Services/SettingsBase.cs

```diff
@@ -75,6 +76,16 @@ protected T GetSetting<T>(Func<T> defaultValue, [CallerMemberName] string key =
             return fallbackValue;
         }
 
+        /// <summary>
+        /// Sets a settings value to its default.
+        /// </summary>
+        /// <param name="key">A unique identifier for the setting.</param>
+        /// <typeparam name="T">The type of the stored value.</typeparam>
+        public void ResetSetting<T>(string key)
+        {
+            SetSetting<T>(default!, key);
```

Comment:
The way we're providing `default` but checking for `null` to do the actual reset makes me uneasy about how this would handle a struct or primitive.

Rather than passing this off to `SetSetting<T>()`, how about we do the reset right here? Just remove the value from `_runtimeStorage` and emit `PropertyChanged`. 

That remove any default/null issues and remove the need for the `T` on `ResetSetting<T>(string key)`.

Reply:
I thought about using `_runtimeStorage` directly, but I decided not to in order to limit its direct use to the one function. I see it's not the optimal solution, so I'll change it.